File: src/main/java/com/matag/admin/deck/DeckController.java
package com.matag.admin.deck;

import static java.util.Collections.emptyList;

import java.util.List;

import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/deck")
public class DeckController {

  @PreAuthorize("hasRole('USER')")
  @GetMapping
  public List<Deck> decks() {
    return emptyList();
  }
}

File: src/main/kotlin/com/matag/admin/deck/DeckController.kt
package com.matag.admin.deck

import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/deck")
open class DeckController {
    @PreAuthorize("hasRole('USER')")
    @GetMapping
    open fun decks(): List<Deck> =
        listOf()
}
